### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by yashodhicy**
🥇 **Interface**

🔴    **Create a class Nameable.**

  
  🌸  Implement a method called correct_name that will raise a NotImplementedError.
  🌸 Turn Person class to Nameable
  🌸Person class inherits from Nameable
  🌸Has a method correct_name implemented that returns the name attribute.

🔴  **Prepare base Decorator**

   🌸 Inherits from Nameable.
   🌸 In the constructor assign a nameable object from params to an instance variable.
   🌸 Implement the correct_name method that returns the result of the correct_name method of the @nameable.

🥇**Prepare CapitalizeDecorator and TrimmerDecorator**

🔴  **For the CapitalizeDecorator:**

  🌸 Create a class that inherits from the base Decorator class.
  🌸 Implement a method correct_name that capitalizes the output of @nameable.correct_name.

🔴  **For the TrimmerDecorator:**

  🌸 Create a class that inherits from the base Decorator class.
  🌸 Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './decorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    to_upper_case(@nameable.correct_name)
+  end
+
+  def to_upper_case(name)
+    name.upcase
```
Comment (iambenkis) [inline capitalize_decorator.rb]: - [x] Good job creating the Capitalize class, however, the capitalizer method should capitalize the `name` not to uppercase 😄  😄  😄  it. Kindly update that

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @yashodhicy 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️ TrimmerDecorator 
✔️ Decorate Class 
✔️ CapitalizeDecorator Class

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: capitalize_decorator.rb (hunk 1 above)
Comment (yashodhicy) [reply capitalize_decorator.rb]: thank  you so much for catching this

**Review by iambenkis — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

✔️ TrimmerDecorator
✔️ Decorate Class
✔️ CapitalizeDecorator Class

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**